ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -469,6 +469,12 @@ private boolean isEligibleForCompaction(CompactionInfo ci, ShowCompactResponse c
             "=true so we will not compact it.");
         return false;
       }
+      if (AcidUtils.isInsertOnlyTable(t.getParameters()) && !HiveConf
```

Comment:
Does this config makes any sense or just adds extra hassle for end users - "Whether the compactor should compact insert-only tables. A safety switch."?

Reply:
I actually have a customer where this would be very useful. MM compaction makes the Workers to hung indefinitely on their system and since all worker thread stops, compaction is down for every table.

Reply:
but what would happen to insert only tables, they won't be compacted ever?

Reply:
no until the underlying bug is fixed, but it would enable the full acid tables to work properly

Reply:
what kind of underlying bug? in most cases query based compaction is used for full acid tables as well. this config looks like a real hack to ignore the problem that over time could lead to a more serious issue with accumulated data to be compacted and performance degradation. If we want to improve compaction usability I don't think that's a good approach, but that's just my 10 cents. From code point of view patch looks good to me.

Reply:
btw, if needed, they could disable auto-compaction on table level

Reply:
1. This switch is already there it will just cause ClassCastException, rather than handle this properly
2. This customer is using MR based compaction which works fine, the problem is for mm tables the system will run query based no matter what, we want to turn off query based compaction for every table, until they can get the fix to their cluster, which is a very valid thing to do, otherwise compaction will not run for anything.
3. Using a switch like this until the upgrade is much better solution then disabling it off for 1000+ tables manually

Reply:
Just my thought on that, feel free to ignore them:
If something is already there doesn't mean it's a good practice. The presence of a switch just indicates how unreliable compaction functionality is that we still need to maintain feature flag. 
Do we have similar toggle just for MR based compaction? :)
Until the upgrade they should probably avoid using MM tables.

Reply:
@deniskuzZ  I guess the original intent for this config was as a feature flag. Since compaction _is_ unreliable at the moment I don't think it's a tragedy if we keep it for a bit longer. Why don't we open a ticket for its removal and do it when we feel ready?

Reply:
@klcopp, that sounds good! My point was that once compaction is stable, we should probably revisit some of these toggles.